components/org.wso2.carbon.identity.api.user.common/src/main/java/org/wso2/carbon/identity/api/user/common/ContextLoader.java

```diff
@@ -221,13 +231,25 @@ public static URI buildURIForHeader(String endpoint) {
      * in non tenant qualified mode we need to append the tenant domain to the path manually.
      *
      * @param endpoint Relative endpoint path.
+     * @param isServerResource Is service URL builds for server resource.
      * @return Context of the API.
      */
-    private static String getContext(String endpoint) {
+    private static String getContext(String endpoint, boolean isServerResource) {
 
         String context;
+        String organizationId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId();
         if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
-            context = USER_API_PATH_COMPONENT + endpoint;
+            if (isServerResource) {
+                context = SERVER_API_PATH_COMPONENT + endpoint;
+            } else {
+                context = USER_API_PATH_COMPONENT + endpoint;
+            }
+            if (StringUtils.isNotEmpty(organizationId)) {
+                String tenantDomain = (String) IdentityUtil.threadLocalProperties.get()
+                        .get(OrganizationManagementConstants.ROOT_TENANT_DOMAIN);
+                context = String.format(TENANT_CONTEXT_PATH_COMPONENT, tenantDomain) +
+                        ORGANIZATION_CONTEXT_PATH_COMPONENT + context;
+            }
```

Comment:
This logic may not required if this logic is moved to the service URL builder logic in the framework

Reply:
Fixed by - https://github.com/wso2/identity-api-user/pull/187